test/public/runs/detail.test.js

```diff
@@ -505,4 +506,57 @@ module.exports = () => {
             innerText: 'ECS',
         });
     });
+
+    it('should display correct tag styling after updating in tag overview', async () => {
+        /**
+         *  Retrieve the badge classes and styles
+         *
+         *  @return {Promise<Array>} resolves with the badge classes and styles
+         */
+        const getRunTagsBadges = async () => {
+            // Check if the tag is updated
+            const tagsBadgeClassesSelector = '#Run-tags .badge';
+            // Wait for badge elements to appear
+            await page.waitForSelector(tagsBadgeClassesSelector);
+            // Evaluate and check for inline background color
+            return await page.$$eval(
+                tagsBadgeClassesSelector,
+                (badges) => badges.map((badge) => ({
+                    backgroundColor: badge.style.backgroundColor,
+                })),
+            );
+        };
+
+        let badges;
+        const expectedBgColorBefore = 'rgb(238, 238, 238)'; //Gray
+        const expectedBgColorAfter = 'rgb(255, 0, 0)'; //Red
+
+        // Fetch the run data before update of tag
+        await goToRunDetails(page, 106);
+
+        badges = await getRunTagsBadges();
+
+        expect(badges[0].backgroundColor == expectedBgColorBefore).to.be.true;
```

Comment:
```suggestion
        expect(badges[0]).to.equal(expectedBgColorBefore);
```

Doing so, you have explicit error message when values are not equal, something like `expected rgb(...) to equal rgb(...)`
When you expect a value of an array, putting an expect on the array length is often a good idea

Reply:
Done